csharp/autobuilder/Semmle.Autobuild.CSharp/DotNetRule.cs

```diff
@@ -79,10 +79,10 @@ public BuildScript Analyse(IAutobuilder<CSharpAutobuildOptions> builder, bool au
         /// variables needed by the installed .NET Core (<code>null</code> when no variables
         /// are needed).
         /// </summary>
-        public static BuildScript WithDotNet(IAutobuilder<AutobuildOptionsShared> builder, Func<string?, IDictionary<string, string>?, BuildScript> f)
+        public static BuildScript WithDotNet(IAutobuilder<AutobuildOptionsShared> builder, bool ensureDotNetAvailable, Func<string?, IDictionary<string, string>?, BuildScript> f)
         {
             var installDir = builder.Actions.PathCombine(FileUtils.GetTemporaryWorkingDirectory(builder.Actions.GetEnvironmentVariable, builder.Options.Language.UpperCaseName, out var _), ".dotnet");
-            var installScript = DownloadDotNet(builder, installDir);
+            var installScript = DownloadDotNet(builder, installDir, ensureDotNetAvailable);
             return BuildScript.Bind(installScript, installed =>
             {
                 Dictionary<string, string>? env;
```

Comment:
Maybe create the dictionary here and insert the shared key/value pairs for both branches (`DOTNET_SKIP_FIRST_TIME_EXPERIENCE` and `MSBUILDDISABLENODEREUSE`)